crates/hyperdrive-math/src/long/targeted.rs

```diff
@@ -0,0 +1,503 @@
+                    "get_targeted_long: Unable to find an acceptible loss. Final loss = {}.",
+                    loss
+                ));
+            }
+
+            Ok(possible_target_base_delta)
+        }
+    }
+
+    /// The fixed rate after a long has been opened.
+    ///
+    /// We calculate the rate for a fixed length of time as:
+    /// $$
+    /// r(x) = (1 - p(x)) / (p(x) t)
+    /// $$
+    ///
+    /// where $p(x)$ is the spot price after a long for `delta_bonds`$= x$ and
+    /// t is the normalized position druation.
+    ///
+    /// In this case, we use the resulting spot price after a hypothetical long
+    /// for `base_amount` is opened.
+    fn rate_after_long(
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: Option<FixedPoint>,
+    ) -> Result<FixedPoint> {
+        let resulting_price = self.calculate_spot_price_after_long(base_amount, bond_amount)?;
+        Ok((fixed!(1e18) - resulting_price)
+            / (resulting_price * self.annualized_position_duration()))
+    }
+
+    /// The derivative of the equation for calculating the rate after a long.
+    ///
+    /// For some $r = (1 - p(x)) / (p(x) * t)$, where $p(x)$
+    /// is the spot price after a long of `delta_base`$= x$ was opened and $t$
+    /// is the annualized position duration, the rate derivative is:
+    ///
+    /// $$
+    /// r'(x) = \frac{(-p'(x) p(x) t - (1 - p(x)) (p'(x) t))}{(p(x) t)^2} //
+    /// r'(x) = \frac{-p'(x)}{t p(x)^2}
+    /// $$
+    ///
+    /// We return $-r'(x)$ because negative numbers cannot be represented by FixedPoint.
+    fn rate_after_long_derivative_negation(
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: FixedPoint,
+    ) -> Result<FixedPoint> {
+        let price = self.calculate_spot_price_after_long(base_amount, Some(bond_amount))?;
+        let price_derivative = self.price_after_long_derivative(base_amount, bond_amount)?;
+        // The actual equation we want to represent is:
+        // r' = -p' / (t p^2)
+        // We can do a trick to return a positive-only version and
+        // indicate that it should be negative in the fn name.
+        Ok(price_derivative / (self.annualized_position_duration() * price.pow(fixed!(2e18))))
+    }
+
+    /// The derivative of the price after a long.
+    ///
+    /// The price after a long that moves shares by $\Delta z$ and bonds by $\Delta y$
+    /// is equal to $P(\Delta z) = \frac{\mu (z_e + \Delta z)}{y - \Delta y}^T$,
+    /// where $T$ is the time stretch constant and $z_e$ is the initial effective share reserves.
+    /// Equivalently, for some amount of `delta_base`$= x$ provided to open a long,
+    /// we can write:
+    ///
+    /// $$
+    /// p(x) = \frac{\mu (z_e + \frac{x}{c} - g(x) - \zeta)}{y_0 - Y(x)}^{T}
+    /// $$
+    /// where $g(x)$ is the [open_long_governance_fee](long::fees::open_long_governance_fee),
+    /// $Y(x)$ is the [long_amount](long::open::calculate_open_long), and $\zeta$ is the
```

Comment:
We're opening a long, so we know that the zeta adjustment will be zero. I'd suggest removing this here and in the following calculations to simply the comments.

Reply:
I'd also say that $\zeta$ is already factored into $z_e$. Maybe I'm missing something, but I don't expect to see $\zeta$ here. $\Delta \zeta$ would make more sense, but it will be zero.

Reply:
Thanks for pointing out this potential source of confusion, I added a comment clarifying $\Delta \zeta$ and I updated the targeted long PDF.

I just went over it again and I believe these equations are correct. The price is computed from the effective share reserves, and the equations are derived wrt the change in shares, $\Delta z$, which is not taking into account the zeta adjustment. Therefore, the equations for $p(x)$ and $p'(x)$ will want to require $z_{e,1} = z_{e,0} + (\Delta z - (\zeta + \Delta \zeta))$. In other words,

$$
z_1 = z_0 + \frac{
    \Delta x - \phi_g \cdot \phi_c \cdot \left( 1 - p \right) \cdot \Delta x
}{c}
$$

$$
\zeta_1 = \zeta_0
$$

and $z_{e,1} = z_{1} - \zeta_{1}$. Therefore,

$$
z_{e,1} = z_0 + \frac{
    \Delta x - \phi_g \cdot \phi_n \cdot \left( 1 - p \right) \cdot \Delta x
}{c} - \zeta_{1}
$$



I agree keeping $\zeta$ around is cumbersome because we're not changing it. However, if a user follows the `calculate_spot_price_after_long` math all the way into `yield_space::calculate_spot_price` they will see that we use `self.ze()`, which computes $z - \zeta$. So leaving it out would be inaccurate.

Reply:
Ahh I think I just got it. This:

$$
z_{e,1} = z_0 + \frac{
    \Delta x - \phi_g \cdot \phi_n \cdot \left( 1 - p \right) \cdot \Delta x
}{c} - \zeta_{1}
$$

is correct, but not what I'm implementing in code. The $- \zeta$ part happens inside `self.ze()`, as I said, and if I do it again here then we subtract it twice.